### PR TITLE
Null check on not found collection: issue #33

### LINKED ISSUE #33: Difficulty with passThruUnknownUrl
I'm new to angular and all these, so I'm kinda overwhelmed with whatever is happening. Everything works fine when only mock data are being accessed when using in-memory-web-api. But when trying to use mock data as specified on https://github.com/angular/in-memory-web-api/issues/30, these errors pop up on the browser console(it's kinda lengthy).

I'm using InMemoryWebApiModule.forRoot(DepartmentData, { passThruUnknownUrl: true }).

https://gist.github.com/kevin-madhu/09f40474b2d648140f0f93d7281ee04e

### PR DESCRIPTION
**Pull request by psamsotha**
Fixes #33 

Two changes, discussed [here][1]

[1]: https://github.com/angular/in-memory-web-api/issues/33#issuecomment-250078603

### CODE REVIEW

**Comment by psamsotha** [comment]
I signed it!

File: src/in-memory-backend.service.ts [hunk 1]
```diff
@@ -437,7 +437,7 @@ export class InMemoryBackendService {
   // tries to parse id as number if collection item.id is a number.
   // returns the original param id otherwise.
   protected parseId(collection: {id: any}[], id: string): any {
-    if (!id) { return null; }
+    if (!collection || !id) { return null; }
```
Comment (wardbell) [inline src/in-memory-backend.service.ts]: Yes to this one.

File: src/in-memory-backend.service.ts [hunk 2]
```diff
@@ -220,7 +220,7 @@ export class InMemoryBackendService {
       if ('commands' === reqInfo.base.toLowerCase()) {
         return this.commands(reqInfo);
 
-      } else if (this.inMemDbService[reqMethodName]) {
+      } else if (this.inMemDbService[reqMethodName] && reqInfo.collection) {
```
Comment (wardbell) [inline src/in-memory-backend.service.ts]: No this this one. The override method should run whether `reqInfo.collection` is present or not.

**Comment by wardbell** [comment]
Closing because PR #39 handles the approved change

**Pull request closed without merging**